subsys/bluetooth/controller/ll_sw/ull_adv_sync.c

```diff
@@ -573,6 +757,28 @@ uint8_t ull_adv_sync_pdu_set_clear(struct ll_adv_set *adv,
 	}
 #endif /* CONFIG_BT_CTLR_DF_ADV_CTE_TX */
 
+#if defined(CONFIG_BT_CTLR_ADV_PDU_LINK)
+	/*
+	 * alloc() will return the same PDU as peek() in case there was PDU
+	 * queued but not switched to current before alloc() - no need to deal
+	 * with chain as it's already there. In other case we need to duplicate
+	 * chain from current PDU and append it to new PDU.
+	 */
+	if (pdu_prev != pdu_new) {
```

Comment:
> Added a couple of minor comments. There are still unresolved comments. Also there are missing following things I've suggested as part of PR I've provided to your private fork:
> 
>     * missing possibility to configure max length of chain ,

max length of chain is determined by max length of contents which is implicitly determined by max adv data - that setting is already configurable so I don't think explicit limit on chain length makes sense since it only creates extra problems. you may want to limit numer of PDUs with CTEInfo, but that is not something to resolve in this PR.

>     * `PDU_MEM_COUNT_MIN` and `PDU_MEM_FIFO_COUNT` hasn't got additional storage for PDUs for chains,

those are already calculated using max adv data and max fragment length so reserve extra space for PDUs in chains. again, you may want to reserve extra space for PDUs with CTEInfo, but that is not to be handled in this PR.

>     * missing possibility to remove single linked PDU, only `lll_adv_pdu_linked_release_from_ull` that releases whole chain from its head.

I'll add separate call